fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/util/Taggable2ObjectMap.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.content.registry.v1.util;
+
+import net.minecraft.tag.Tag;
+
+public interface Taggable2ObjectMap<K, V> {
```

Comment:
Any reason this doesn't just extend `Map<K, V>`?

Reply:
It's really not a map, it's like an interface for a class that interacts with maps. Not sure what a better name is, open to suggestions

Reply:
Is this targetting 20w14a or later? If yes we should not take `Tag<K>` but `Tag.Identified<K>` as that is the persistent form of tag now.

Reply:
I suggest the name `ObjectValue`, like java has a [`ClassValue`](https://docs.oracle.com/javase/8/docs/api/java/lang/ClassValue.html) which is a class key -> value mapping

Reply:
This targets 1.15, where Tag.Identified is not a thing. It won't be hard to change it in 1.16. Don't think ObjectValue makes sense because it is something that lets you add, remove, and get values, but isn't a value itself.